### PR TITLE
Feature/waypoint block

### LINKED ISSUE #1263: Create waypoint placeholder block
This is a placerholder block like the solid one (texture this time blue).

The difference is that it lets the player create a waypoint on building it.

Advantages:

If we add them in all our worker buildings we can remove the waypoint of the building chests.
(This time workers don't enter the buildings).

This way also the bridges etc can have the waypoint in a central pos.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1263

# Changes proposed in this pull request:
- Adds miner model by D3miurge
- Adds a new waypoint block which will be replaced by air and set a waypoint at the location.
We can use this in future buildings to eventually remove the automatic waypoint by colony buildings and have a specific waypoint at the building (entrance for example)

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * Used to determine ambient occlusion and culling when rebuilding chunks
+     * for render.
+     * @deprecated
+     *
+     * @return true
+     */
+    //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:73]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    @SuppressWarnings(DEPRECATION)
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:77]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 3]
```diff
@@ -399,6 +403,23 @@ public Object handle(
...
+            {
+                return ActionProcessingResult.IGNORE;
+            }
+
+            placer.addWayPoint(pos);
+            return blockState;
+        }
+    }
+
     public static class ChestPlacementHandler implements IPlacementHandler
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:423]: MAJOR Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 4]
```diff
@@ -6,6 +6,8 @@
 import com.minecolonies.api.util.ItemStackUtils;
 import com.minecolonies.coremod.blocks.BlockMinecoloniesRack;
 import com.minecolonies.coremod.blocks.BlockSolidSubstitution;
+import com.minecolonies.coremod.blocks.BlockWaypoint;
+import com.minecolonies.coremod.blocks.ModBlocks;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:10]: MINOR Remove this unused import 'com.minecolonies.coremod.blocks.ModBlocks'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java [hunk 5]
```diff
@@ -32,6 +32,8 @@
     public static final Block blockConstructionTapeCorner = new BlockConstructionTapeCorner();
     public static final Block blockTimberFrame            = new BlockTimberFrame();
     public static final Block blockRack                   = new BlockMinecoloniesRack();
+    public static final Block blockWayPoint               = new BlockWaypoint();
+
 
     // Deactivated for now
     // public static final Block blockBarrel        = new BlockBarrel();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java:39]: MINOR This block of commented-out lines of code should be removed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 47 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 36 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L782](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L782): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L66](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L66): Reduce this class from 50 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L68](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L68): Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L236](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L236): Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L100](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L100): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L106](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L106): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L125](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L125): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/42e9f166af6f3f42b8cc313299d04d8a2217149c/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
I would like to add that for now.
We can then, eventually remove the waypoints of the huts as soon as we rescanned all the schematics.

**Comment by Asherslab** [comment]
:shipit: looks good!

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1387/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
@OrionDevelopment

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 0.0F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockwaypoint";
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java]: Could we follow the normal naming convention here? "blockWayPoint" instead? (All other blocks are named that way AFAIK)

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /**
+     * initialize the block
+     * sets the creative tab, as well as the resistance and the hardness.
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(Locale.ENGLISH), BLOCK_NAME));
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:58]: what does the Locale.English part do? (haven't seen it anywhere else)

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        return false;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SuppressWarnings(DEPRECATION)
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:88]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 36 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L782](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L782): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L66](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L66): Reduce this class from 50 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L68](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L68): Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L236](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L236): Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L100](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L100): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L106](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L106): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L125](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L125): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/c5eb21902471dd9482a23ce4c89b68531f2f1e14/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Asherslab** [comment]
Well, it looks great to me, if you could just answer or fix my two comments, lol =P

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java (hunk 6 above)
Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java]: Yep please change.

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java (hunk 7 above)
Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:58]: Nothing as Locale is not set in minecraft, and it then defaults to English.

**Review by marchermans — CHANGES_REQUESTED** [review]
Change the block name. Then we are good to go.

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java (hunk 7 above)
Comment (Asherslab) [reply src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:58]: ahh, cool

**Review by marchermans — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 36 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L782](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L782): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L66](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L66): Reduce this class from 50 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L68](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L68): Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L236](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L236): Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L100](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L100): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L106](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L106): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L125](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L125): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/5ab2e14c0eee42120d740723b954fbbbafce570b/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**